homeassistant/components/varta_storage/__init__.py

```diff
@@ -0,0 +1,252 @@
+"""The VARTA Storage integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+
+import async_timeout
+from vartastorage import vartastorage
+
+from homeassistant import config_entries, core
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ENERGY_KILO_WATT_HOUR,
+    ENERGY_WATT_HOUR,
+    PERCENTAGE,
+    POWER_WATT,
+    TIME_HOURS,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import _LOGGER, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up VARTA Storage from a config entry."""
+    varta = vartastorage.VartaStorage(entry.data["host"], entry.data["port"])
+    try:
+        varta.client.connect()
+    except Exception as ex:
```

Comment:
This is a very broad exception that is being captured, can we make this more specific?

Reply:
done